lib/src/main/kotlin/com/swmansion/starknet/data/serializers/ContractClassGatewaySerializer.kt

```diff
@@ -0,0 +1,37 @@
+package com.swmansion.starknet.data.serializers
+
+import com.swmansion.starknet.data.types.ContractClass
+import com.swmansion.starknet.extensions.base64Gzipped
+import kotlinx.serialization.KSerializer
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.descriptors.SerialDescriptor
+import kotlinx.serialization.encoding.Decoder
+import kotlinx.serialization.encoding.Encoder
+import kotlinx.serialization.json.JsonElement
+
+internal object ContractClassGatewaySerializer : KSerializer<ContractClass> {
+    @Serializable
+    data class ContractClassGateway(
+        val program: JsonElement,
+
+        @SerialName("entry_points_by_type")
+        val entryPointsByType: ContractClass.EntryPointsByType,
+    )
+
+    override fun deserialize(decoder: Decoder): ContractClass {
+        val response = ContractClassGateway.serializer().deserialize(decoder)
+        val programString = response.program.toString()
+        val program = programString.base64Gzipped()
+        // FIXME: It doesn't produce the same output as the rpc endpoint
```

Comment:
Gateway returns in different format than rpc, that's why this serializer was needed I assume. This code was actually not changed but rather moved from types/Contracts.kt. I tried to adapt it to rpc but there were issues.

Reply:
At this point I'm not 100% sure but if I remember correctly, the RPC returns compressed version of the program, that, when uncompressed, is a bit different that the one from gateway. 

It would require verification though.